test/e2e/device_test.go

```diff
@@ -0,0 +1,223 @@
+				if err != nil {
+					t.Error("Error reading table data:", err)
+				}
+
+				// Assuming the ID column is at index 0 (first column)
+				if len(records) > 2 && len(records[1]) > 0 {
+					firstID = records[1][0]
+					t.Error("First ID hash:", firstID)
+				} else {
+					t.Error("No data found in the table.")
+				}
+			},
+		},
+		{
+			name: "get",
+			fields: fields{
+				MainCmd:  devices.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				root := c.Root()
+				root.SetArgs([]string{subCommand, "get"})
+				rescueStdout := os.Stdout
+				r, w, _ := os.Pipe()
+				os.Stdout = w
+				if err := root.Execute(); err != nil {
+					t.Error(err)
+				}
+				w.Close()
+				out, _ := io.ReadAll(r)
+				os.Stdout = rescueStdout
+				if !strings.Contains(string(out[:]), "metal-test-stag1") &&
+					!strings.Contains(string(out[:]), "Ubuntu 20.04 LTS") &&
+					!strings.Contains(string(out[:]), "active") {
+					t.Error("expected output should include metal-test-stag1, Ubuntu 20.04 LTS, and active strings in the out string ")
+				}
+			},
+		},
+		{
+			name: "get_by_device_id",
+			fields: fields{
+				MainCmd:  devices.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				root := c.Root()
+				root.SetArgs([]string{subCommand, "get", "-i", firstID})
+				rescueStdout := os.Stdout
+				r, w, _ := os.Pipe()
+				os.Stdout = w
+				if err := root.Execute(); err != nil {
+					t.Error(err)
+				}
+				w.Close()
+				out, _ := io.ReadAll(r)
+				os.Stdout = rescueStdout
+				if !strings.Contains(string(out[:]), "metal-test-stag1") &&
+					!strings.Contains(string(out[:]), "Ubuntu 20.04 LTS") &&
+					!strings.Contains(string(out[:]), firstID) &&
+					!strings.Contains(string(out[:]), "active") {
+					t.Error("expected output should include metal-test-stag1, Ubuntu 20.04 LTS, and active strings in the out string ")
+				}
+			},
+		},
+		{
+			name: "reboot",
+			fields: fields{
+				MainCmd:  devices.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				root := c.Root()
+				root.SetArgs([]string{subCommand, "reboot", "--id", firstID})
+				rescueStdout := os.Stdout
+				r, w, _ := os.Pipe()
+				os.Stdout = w
+				if err := root.Execute(); err != nil {
+					t.Error(err)
+				}
+				w.Close()
+				out, _ := io.ReadAll(r)
+				os.Stdout = rescueStdout
+				if !strings.Contains(string(out[:]), "Device"+firstID+" successfully rebooted.") {
+					t.Error("expected output should include Device" + firstID + " successfully rebooted.")
```

Comment:
To string together these tests, we'll need to have each test after the `create` test wait for the test device to reach a certain status before attempting to modify it.  That would probably look a bit like this function from the terraform device resource: https://github.com/equinix/terraform-provider-equinix/blob/main/equinix/resource_metal_device.go#L846-L866

Reply:
updtaed